code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -123,3 +123,98 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
 	if(CHECK_BITFIELD(direction_flag, CONE_PART_MIDDLE_DIAG))
 		do_acid_cone_spray(next_normal_turf, distance_left - 1, facing, CONE_PART_DIAG_LEFT|CONE_PART_DIAG_RIGHT, spray)
 		do_acid_cone_spray(next_normal_turf, distance_left - 2, facing, (distance_left < 5) ? CONE_PART_MIDDLE : CONE_PART_MIDDLE_DIAG, spray)
+
+// ***************************************
+// *********** Acid dash
+// ***************************************
+/datum/action/xeno_action/activable/acid_dash
+	name = "Acid Dash"
+	action_icon_state = "charge"
+	mechanics_text = "Instantly dash, tackling the first marine in your path. If you manage to tackle someone, gain another weaker cast of the ability."
+	ability_name = "acid dash"
+	plasma_cost = 250
+	cooldown_timer = 30 SECONDS
+	///How far can we dash
+	var/range = 5
+	///Can we use the ability again
+	var/recast_available = FALSE
+	///Is this the recast
+	var/recast = FALSE
+	///The last tile we dashed through, used when swapping with a human
+	var/turf/last_turf
+
+/datum/action/xeno_action/activable/acid_dash/on_cooldown_finish()
+	to_chat(owner, span_xenodanger("Our exoskeleton quivers as we get ready to use Acid Dash again."))
+	playsound(owner, "sound/effects/xeno_newlarva.ogg", 50, 0, 1)
+	return ..()
+
+///Called when the dash is finished, handles cooldowns and recast. Clears signals too
+/datum/action/xeno_action/activable/acid_dash/proc/dash_complete()
+	SIGNAL_HANDLER
+	if(recast_available)
+		addtimer(CALLBACK(src, .proc/dash_complete), 2 SECONDS) //Delayed recursive call, this time you won't gain a recast so it will go on cooldown in 2 SECONDS.
+		recast = TRUE
+	else
+		recast = FALSE
+		add_cooldown()
+
+	recast_available = FALSE
+	UnregisterSignal(owner, list(COMSIG_XENO_OBJ_THROW_HIT, COMSIG_XENO_NONE_THROW_HIT, COMSIG_XENO_LIVING_THROW_HIT, COMSIG_MOVABLE_MOVED))
+
+///Called whenever the owner hits a mob during the dash
+/datum/action/xeno_action/activable/acid_dash/proc/mob_hit(datum/source, mob/M)
+	SIGNAL_HANDLER
+	if(recast || !ishuman(M)) //That's the recast, we don't stop for mobs
```

Comment:
if it's a recast you don't paralyse and you don't swap?

Reply:
yes you don't